DFE-Digital/sts-plan-technology-for-your-school
# Feat: Invalidate web app cache on content update

## Features

- Adds endpoint to the web app which invalidates the database cache 
- Endpoint requires api key authentication
- Adds functionality to the azure function to call this endpoint on content update
- The API key and Endpoint to call are both stored in the keyvault

## Changes made in Azure (needs replicating in other envs)

- Added `CacheClear--Endpoint` and `CacheClear--ApiKey` to the secrets in the dev keyvault
- Added `WEBSITE_CACHE_CLEAR_ENDPOINT` and `WEBSITE_CACHE_CLEAR_APIKEY` to the Environment variables of the function app in dev, pointing to the secrets

## How to test Locally

1. Disable The QueueReciever in Azure Portal
2. Go to the Keyvault and use the following secrets to update your local configuration
    * Update your dotnet secrets for PlanTech.Web and add:
        * `"CacheClear:Endpoint": "<endpoint from keyvault>"`,
        * `"CacheClear:ApiKey": "<apikey from keyvault>"`
    * Update your local.settings.json file for the Azure function and add:
        * `"WEBSITE_CACHE_CLEAR_ENDPOINT":  "<endpoint from keyvault>"`,
        * `"WEBSITE_CACHE_CLEAR_APIKEY": "<apikey from keyvault>"`
3. Run the web app and the azure function 
4. Go to a page in the web app
5. Change the content of it in contentful dev
6. Refresh that page, and the content should be updated, because the cache got cleared.
7. Validate that this is *Not* the case if you change the API keys in your secrets/local settings to be different/missing
